### DISCUSSION TITLE
Using atoms as default for internal state.

### DISCUSSION THREAD

**Discussion by geoffreydhuyvetters** [pmndrs/jotai#268] [Q&A] [answered]
Does it make sense to default to useAtom instead of useState in a new application? 
Or does that come with a some performance overhead?

**Comment by dai-shi** [comment, marked as answer]
Well, it certainly has some overhead. For local state, where useState and prop drilling make sense, useState would be recommended (not only for performance but also for component scope locality).

That said, the overhead is minimal, and you wouldn't notice unless you measure it or your app is very heavy. For toy projects, using `useAtom` everywhere might be ok. It could increase developer experience by isolating logic out of components with unified pattern (atoms).

**Reply by geoffreydhuyvetters** [reply]
Makes a lot of sense. Thanks for the swift reply!
